substrate/frame/treasury/src/lib.rs

```diff
@@ -451,17 +462,27 @@ pub mod pallet {
 			}
 
 			// Check to see if we should spend some funds!
-			if (n % T::SpendPeriod::get()).is_zero() {
-				Self::spend_funds()
+			let last_spend_period =
+				// AUDIT REVIEW TODO! Needs to handle migration story for this better.
+				LastSpendPeriod::<T, I>::get().unwrap_or(BlockNumberFor::<T>::zero());
```

Comment:
call out here

probably needs migration, but perhaps there is a way to avoid it with some clever logic?

Reply:
```suggestion
				LastSpendPeriod::<T, I>::get().unwrap_or_else(|| calculate_last_spend_period());
```

in that way no migration is needed if the block number provider is the local block number

but we will need a migration when switch the block number provider

Reply:
> but we will need a migration when switch the block number provider

Seems unavoidable